python/Phdlcpp.cpp

```diff
@@ -1,16 +1,19 @@
 #include <pybind11/pybind11.h>
 #include <pybind11/functional.h>
+#include <memory>
 #include "../include/Hdlcpp.hpp"
 
 using Pybind11Read = std::function<char(int)>;
 using Pybind11Write = std::function<int(pybind11::bytes)>;
 
+using Hdlcpp_t = Hdlcpp::Hdlcpp<256>;
```

Comment:
Any thoughts on the static buffer size for the python wrapper? - ideas to allow for user controlled buffer size?

Reply:
Maybe this: https://pybind11.readthedocs.io/en/stable/advanced/functions.html#binding-functions-with-template-parameters

Reply:
It seems to be only for template functions, and with a predefined type for each function.

I think we should add the size used for the python binding in CMake, then it is visible and easily changed when building the python module.

Reply:
https://github.com/bang-olufsen/hdlcpp/blob/master/python/hdlcpp.py#L5 defines the size. Maybe add a `static assert` to ensure that the size is as provided / for compatibility.

Reply:
Already removed the default buffer size argument for runtime
https://github.com/bang-olufsen/hdlcpp/pull/33/files#diff-6d2b38968d2e45aa339011797c90373d42d6823edfcc491e11f0d619d6e72ecfL5
New python constructor:
```python
class Hdlcpp:
    def __init__(self, transportRead, transportWrite, writeTimeout=100, writeRetries=1):
        self.hdlcpp = phdlcpp.Hdlcpp(transportRead, transportWrite, writeTimeout, writeRetries)
```

Reply:
> https://github.com/bang-olufsen/hdlcpp/blob/master/python/hdlcpp.py#L5 defines the size. Maybe add a `static assert` to ensure that the size is as provided / for compatibility.

But a `static_assert` that it is not 0 is a really god idea! :smile: